### PR TITLE
Drtii 864 departure

### PR DESCRIPTION
**Pull request by ngu04**
- making departure details to be non mandatory while parsing csv file .

### CODE REVIEW

File: src/main/scala/uk/gov/homeoffice/drt/routes/UploadRoutes.scala [hunk 1]
```diff
@@ -107,12 +107,14 @@ object UploadRoutes extends JsonSupport {
...
-      departureTime = indexMapRow.getOrElse(7, "").trim,
-      embarkPort = indexMapRow.getOrElse(8, "").trim,
-      departurePort = indexMapRow.getOrElse(9, "").trim)
+      departureDate = emptyColumnCheck(indexMapRow.getOrElse(6, "")),
+      departureTime = emptyColumnCheck(indexMapRow.getOrElse(7, "")),
+      embarkPort = emptyColumnCheck(indexMapRow.getOrElse(8, "")),
+      departurePort = emptyColumnCheck(indexMapRow.getOrElse(9, "")))
   }
 
+  val emptyColumnCheck: String => Option[String] = column => if (column.isEmpty) None else Option(column)
```
Comment (richbirch) [inline src/main/scala/uk/gov/homeoffice/drt/routes/UploadRoutes.scala]: This function seems to be redundant if you change `getOrElse` to `get` in `indexMapRow.get(xxx)` would have the same result

File: src/main/scala/uk/gov/homeoffice/drt/routes/UploadRoutes.scala [hunk 2]
```diff
@@ -128,16 +130,16 @@ object UploadRoutes extends JsonSupport {
...
+                      departurePort = flightRowsByArrival.head.departurePort.map(_.trim),
+                      embarkPort = flightRowsByArrival.head.embarkPort.map(_.trim),
                       flightRowsByArrival.size)
                 }
             }
       }.toList
   }
 
-  val covertDateTime: String => MillisSinceEpoch = date => if (date.isEmpty) 0 else
+  val covertDateTime: String => MillisSinceEpoch = date =>
```
Comment (richbirch) [inline src/main/scala/uk/gov/homeoffice/drt/routes/UploadRoutes.scala]: i'd like this function to be be renamed to describe what it does. eg `parseDateToMillis`

**Review by richbirch — APPROVED** [review]
Good stuff. Just a couple of suggestions - a rename and probably function to remove to simplify the code :+1:

File: src/main/scala/uk/gov/homeoffice/drt/routes/UploadRoutes.scala (hunk 1 above)
Comment (ngu04) [reply src/main/scala/uk/gov/homeoffice/drt/routes/UploadRoutes.scala]: One of file load failed as there was " " in the value of column . So this check is needed

Comment (richbirch) [reply src/main/scala/uk/gov/homeoffice/drt/routes/UploadRoutes.scala]: Ok. Let's rename it to something that says what it returns. eg `maybeColumnContent`

**Pull request merged**